<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Run analysis • HaplinMethyl</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cerulean/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Run analysis">
<meta property="og:description" content="HaplinMethyl">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">HaplinMethyl</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">Home</a>
</li>
<li>
  <a href="../articles/index.html">Vignettes</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/jromanowska/HaplinMethyl">
    <span class="glyphicon glyphicon-download"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="Run_analysis_files/header-attrs-2.8/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Run analysis</h1>
                        <h4 class="author">Julia Romanowska</h4>
            
            <h4 class="date">2021-06-29</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/jromanowska/HaplinMethyl/blob/master/vignettes/Run_analysis.Rmd"><code>vignettes/Run_analysis.Rmd</code></a></small>
      <div class="hidden name"><code>Run_analysis.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>This vignette will show you how to run the interaction analyses using <a href="https://cran.r-project.org/package=Haplin"><code>Haplin</code> package</a>.</p>
</div>
<div id="reading-data" class="section level1">
<h1 class="hasAnchor">
<a href="#reading-data" class="anchor"></a>Reading data</h1>
<p>First, we need to read two datasets: DNA methylation and genotype. We’ll use simulated data here.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jromanowska/HaplinMethyl/">HaplinMethyl</a></span><span class="op">)</span>
<span class="co">#&gt; Loading required package: Haplin</span>

<span class="va">ex_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, package <span class="op">=</span> <span class="st">"HaplinMethyl"</span><span class="op">)</span>
<span class="va">ex_file</span> <span class="op">&lt;-</span> <span class="st">"env_data_test.dat"</span>
<span class="va">ex_out_file</span> <span class="op">&lt;-</span> <span class="st">"dnam_ex"</span>

<span class="va">dnam_ex</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/envDataRead.html">envDataRead</a></span><span class="op">(</span>
  file.in <span class="op">=</span> <span class="va">ex_file</span>,
  dir.in <span class="op">=</span> <span class="va">ex_path</span>,
  file.out <span class="op">=</span> <span class="va">ex_out_file</span>,
  sep <span class="op">=</span> <span class="st">" "</span>,
  overwrite <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span>
<span class="co">#&gt; The output file(s) exist!</span>
<span class="co">#&gt; Reading the data in chunks...</span>
<span class="co">#&gt;  -- chunk 1--</span>
<span class="co">#&gt;  -- chunk 2--</span>
<span class="co">#&gt; ... done reading.</span>
<span class="co">#&gt; Preparing data...</span>
<span class="co">#&gt; ... done preparing</span>
<span class="co">#&gt; Saving data...</span>
<span class="co">#&gt; ... saved to file: ./dnam_ex_env.ffData</span>

<span class="va">ex_gen_file</span> <span class="op">&lt;-</span> <span class="st">"sim1"</span>
<span class="va">gen_ex</span> <span class="op">&lt;-</span> <span class="fu">genDataLoad</span><span class="op">(</span>
  filename <span class="op">=</span> <span class="va">ex_gen_file</span>,
  dir.in <span class="op">=</span> <span class="va">ex_path</span>
<span class="op">)</span></code></pre></div>
<p>Let’s take a look at it</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dnam_ex</span>
<span class="co">#&gt; This is continuous environmental data read in by 'envDataRead'</span>
<span class="co">#&gt; with 400 columns</span>
<span class="co">#&gt; and 200 rows.</span>
<span class="fu"><a href="../reference/showRaw.html">showRaw</a></span><span class="op">(</span><span class="va">dnam_ex</span><span class="op">)</span>
<span class="co">#&gt; opening ff /tmp/RtmpDbDIRL/ff/ff44bd96beff9.ff</span>
<span class="co">#&gt;           cg1        cg2        cg3       cg4       cg5</span>
<span class="co">#&gt; id1 0.2045598 0.38376996 0.76001042 0.8731273 0.7625753</span>
<span class="co">#&gt; id2 0.2593365 0.41689855 0.94649631 0.9673046 0.4619886</span>
<span class="co">#&gt; id3 0.9912362 0.91398561 0.07621296 0.3841574 0.7026818</span>
<span class="co">#&gt; id4 0.4740415 0.04124038 0.56624919 0.4589275 0.4461102</span>
<span class="co">#&gt; id5 0.7989879 0.52987689 0.66167605 0.1056240 0.3917910</span>
<span class="co">#&gt; attr(,"Csingle")</span>
<span class="co">#&gt; [1] TRUE</span>

<span class="va">gen_ex</span>
<span class="co">#&gt; This is raw genetic data read in through genDataRead.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; It contains the following parts:</span>
<span class="co">#&gt;    cov.data, gen.data, aux </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; with following dimensions:</span>
<span class="co">#&gt;   - covariate variables =  cov.1 </span>
<span class="co">#&gt;       (total  1  covariate variables),</span>
<span class="co">#&gt;   - number of markers =  1 ,</span>
<span class="co">#&gt;   - number of data lines =  200</span>
<span class="fu">showGen</span><span class="op">(</span><span class="va">gen_ex</span><span class="op">)</span>
<span class="co">#&gt; Warning in showGen(gen_ex): It appears that your data has less markers (1) than</span>
<span class="co">#&gt; requested (5), adjusting accordingly.</span>
<span class="co">#&gt; opening ff /tmp/RtmpDbDIRL/ff/tmp/RtmpekQIls/ff/ff164011e6d830.ff</span>
<span class="co">#&gt; ff (open) byte length=30 (30) dim=c(5,6) dimorder=c(1,2) levels: 1 2</span>
<span class="co">#&gt;      l_m1_a_m l_m1_b_m l_m1_a_f l_m1_b_f l_m1_a_c l_m1_b_c</span>
<span class="co">#&gt; [1,]        1        1        2        1        1        1</span>
<span class="co">#&gt; [2,]        1        1        2        2        2        1</span>
<span class="co">#&gt; [3,]        1        2        1        1        1        2</span>
<span class="co">#&gt; [4,]        1        1        2        2        2        1</span>
<span class="co">#&gt; [5,]        1        2        2        2        2        1</span></code></pre></div>
<p>The genotype in the example is coded as 1 and 2 instead of a specific base letter (e.g., A and T, or A and C) and there is only one marker. The order of the columns is as follows: maternal allele 1 (A1), maternal A2, paternal A1, paternal A2, child’s A1, and child’s A2.</p>
</div>
<div id="creating-the-strata" class="section level1">
<h1 class="hasAnchor">
<a href="#creating-the-strata" class="anchor"></a>Creating the strata</h1>
<p>Let’s assume that we know that the first three methylation sites are near the marker that is in the genotype dataset. Thus, we need to extract those methylation sites first.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dnam_ex_subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/envDataSubset.html">envDataSubset</a></span><span class="op">(</span>
  env.data <span class="op">=</span> <span class="va">dnam_ex</span>,
  col.ids <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">3</span>,
  file.out <span class="op">=</span> <span class="st">"dnam_subset"</span>
<span class="op">)</span>
<span class="co">#&gt; Will select 3 columns.</span>
<span class="co">#&gt; Saving data... </span>
<span class="co">#&gt; ... saved to files: ./dnam_subset_gen.ffData, ./dnam_subset_gen.RData</span>
<span class="va">dnam_ex_subset</span>
<span class="co">#&gt; This is continuous environmental data read in by 'envDataRead'</span>
<span class="co">#&gt; with 3 columns</span>
<span class="co">#&gt; and 200 rows.</span></code></pre></div>
<p>Next, let’s create one variable that gives division of the samples into strata based on the <em>summarized</em> methylation level over the three chosen methylation sites.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dnam_ex_cat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/envDataCategorize.html">envDataCategorize</a></span><span class="op">(</span>
  env.data <span class="op">=</span> <span class="va">dnam_ex_subset</span>,
  breaks <span class="op">=</span> <span class="fl">3</span>,
  file.out <span class="op">=</span> <span class="st">"dnam_cat"</span>
<span class="op">)</span>
<span class="co">#&gt; opening ff /tmp/RtmpDbDIRL/ff/ff44bd59f1510f.ff</span>
<span class="co">#&gt; Creating categories: 1,2,3</span>
<span class="co">#&gt; Saving data...</span>
<span class="co">#&gt; ... saved to file: ./dnam_cat_gen.ffData</span>
<span class="va">dnam_ex_cat</span>
<span class="co">#&gt; This is categorical environmental data read in by 'envDataRead'</span>
<span class="co">#&gt; with 1 columns</span>
<span class="co">#&gt; and 200 rows.</span>
<span class="fu"><a href="../reference/showRaw.html">showRaw</a></span><span class="op">(</span><span class="va">dnam_ex_cat</span><span class="op">)</span>
<span class="co">#&gt; opening ff /tmp/RtmpDbDIRL/ff/ff44bd110a5741.ff</span>
<span class="co">#&gt;     [,1]</span>
<span class="co">#&gt; id1 2   </span>
<span class="co">#&gt; id2 2   </span>
<span class="co">#&gt; id3 3   </span>
<span class="co">#&gt; id4 1   </span>
<span class="co">#&gt; id5 3   </span>
<span class="co">#&gt; attr(,"vmode")</span>
<span class="co">#&gt; [1] byte</span>
<span class="co">#&gt; Levels: 1 2 3</span></code></pre></div>
</div>
<div id="adding-the-strata-variable" class="section level1">
<h1 class="hasAnchor">
<a href="#adding-the-strata-variable" class="anchor"></a>Adding the strata variable</h1>
<p>Now, the strata variable needs to be added to the genotype dataset.</p>
<blockquote>
<p>NB: be careful to check that the order of the samples is the same in the stratified DNA methylation data as in the genotype data!</p>
</blockquote>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">new_strata</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/showRaw.html">showRaw</a></span><span class="op">(</span><span class="va">dnam_ex_cat</span>, rows <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="../reference/nrows.html">nrows</a></span><span class="op">(</span><span class="va">dnam_ex_cat</span><span class="op">)</span><span class="op">)</span>
<span class="va">gen_ex_strat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/addCovar.html">addCovar</a></span><span class="op">(</span>
  data.in <span class="op">=</span> <span class="va">gen_ex</span>,
  covar <span class="op">=</span> <span class="va">new_strata</span>,
  c.name <span class="op">=</span> <span class="st">"dnam_c"</span>
<span class="op">)</span>
<span class="va">gen_ex_strat</span>
<span class="co">#&gt; This is raw genetic data read in through genDataRead.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; It contains the following parts:</span>
<span class="co">#&gt;    cov.data, gen.data, aux </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; with following dimensions:</span>
<span class="co">#&gt;   - covariate variables =  cov.1, dnam_c </span>
<span class="co">#&gt;       (total  2  covariate variables),</span>
<span class="co">#&gt;   - number of markers =  1 ,</span>
<span class="co">#&gt;   - number of data lines =  200</span>
<span class="fu">showPheno</span><span class="op">(</span><span class="va">gen_ex_strat</span><span class="op">)</span>
<span class="co">#&gt;     cov.1 dnam_c</span>
<span class="co">#&gt; id1 "1"   "2"   </span>
<span class="co">#&gt; id2 "1"   "2"   </span>
<span class="co">#&gt; id3 "1"   "3"   </span>
<span class="co">#&gt; id4 "1"   "1"   </span>
<span class="co">#&gt; id5 "1"   "3"</span></code></pre></div>
</div>
<div id="preparing-genotype-data" class="section level1">
<h1 class="hasAnchor">
<a href="#preparing-genotype-data" class="anchor"></a>Preparing genotype data</h1>
<p>Haplin requires pre-processing of the genotype data before any analysis can be made.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gen_ex_strat_prep</span> <span class="op">&lt;-</span> <span class="fu">genDataPreprocess</span><span class="op">(</span>
  data.in <span class="op">=</span> <span class="va">gen_ex_strat</span>,
  file.out <span class="op">=</span> <span class="st">"gen_strat_prep"</span>
<span class="op">)</span>
<span class="co">#&gt; Reading the marker names...</span>
<span class="co">#&gt; Warning: No map file given, map file empty or the number of map file rows not</span>
<span class="co">#&gt; equal to the number of markers in data; will generate dummy marker names.</span>
<span class="co">#&gt; ...done</span>
<span class="co">#&gt; Recoding covariate data...</span>
<span class="co">#&gt; ...done</span>
<span class="co">#&gt; Recoding genetic data (no. of loci: 1)...</span>
<span class="co">#&gt;    ...running on only one CPU core! This may take some time...</span>
<span class="co">#&gt;    ...checking alleles per SNP...</span>
<span class="co">#&gt;    ...done, all alleles: 1 2 </span>
<span class="co">#&gt;    ...recoding SNPs...</span>
<span class="co">#&gt;    ...done</span>
<span class="co">#&gt; Saving data... </span>
<span class="co">#&gt; ... saved to files: ./gen_strat_prep_gen.ffData ,  ./gen_strat_prep_gen.RData</span>
<span class="va">gen_ex_strat_prep</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;      This is preprocessed data, ready for haplin analysis.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;      It contains the following parts:</span>
<span class="co">#&gt;    cov.data, gen.data, aux </span>
<span class="co">#&gt; </span>
<span class="co">#&gt;      with following dimensions:</span>
<span class="co">#&gt;    - number of covariate variables =  2 ,  - number of markers =  1 ,  - number of individuals/families =  200</span></code></pre></div>
</div>
<div id="run-the-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#run-the-analysis" class="anchor"></a>Run the analysis</h1>
<div id="gene-methylation-interactions" class="section level2">
<h2 class="hasAnchor">
<a href="#gene-methylation-interactions" class="anchor"></a>Gene-methylation interactions</h2>
<p>Finally, we can run the analysis by using <code>haplinStrat</code></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">g_x_me_results</span> <span class="op">&lt;-</span> <span class="fu">haplinStrat</span><span class="op">(</span>
  data <span class="op">=</span> <span class="va">gen_ex_strat_prep</span>,
  strata <span class="op">=</span> <span class="fl">2</span>
<span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ## Running haplinStrat ##</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Selected stratification variable: dnam_c.c</span>
<span class="co">#&gt; Frequency distribution of stratification variable:</span>
<span class="co">#&gt;  1  2  3 </span>
<span class="co">#&gt; 67 66 67 </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Running Haplin on full data file...</span>
<span class="co">#&gt; opening ff /tmp/RtmpDbDIRL/ff/ff44bd11e51902.ff</span>
<span class="co">#&gt; Done</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Running Haplin on stratum "1"...Done</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Running Haplin on stratum "2"...Done</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Running Haplin on stratum "3"...Done</span></code></pre></div>
<p>The resulting object is a list of results, with estimates per stratum and for pooled sample.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">g_x_me_results</span><span class="op">)</span>
<span class="co">#&gt; [1] "all" "1"   "2"   "3"</span>
<span class="va">g_x_me_results</span><span class="op">$</span><span class="va">all</span>
<span class="co">#&gt; This is the result of a haplin run.</span>
<span class="co">#&gt; Number of data lines used: 200 | Number of haplotypes used: 2</span>
<span class="co">#&gt; Please use the "summary", "plot", "haptable" or "output" functions to obtain</span>
<span class="co">#&gt;  more details.</span>
<span class="fu">haptable</span><span class="op">(</span><span class="va">g_x_me_results</span><span class="op">)</span>
<span class="co">#&gt;   stratum row.str marker alleles  counts    HWE.pv Original After.rem.NA</span>
<span class="co">#&gt; 1     all       1     m1     1/2 607/593 0.3708224      200          200</span>
<span class="co">#&gt; 2     all       2   &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;        NA      200          200</span>
<span class="co">#&gt; 3       1       1     m1     1/2 178/224 0.4912064       67           67</span>
<span class="co">#&gt; 4       1       2   &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;        NA       67           67</span>
<span class="co">#&gt; 5       2       1     m1     1/2 206/190 0.6861466       66           66</span>
<span class="co">#&gt; 6       2       2   &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;        NA       66           66</span>
<span class="co">#&gt; 7       3       1     m1     1/2 223/179 0.1415450       67           67</span>
<span class="co">#&gt; 8       3       2   &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;        NA       67           67</span>
<span class="co">#&gt;   After.rem.Mend.inc. After.rem.unused.haplos pv.overall haplos haplofreq</span>
<span class="co">#&gt; 1                 200                     200  0.3249067      1 0.5179964</span>
<span class="co">#&gt; 2                 200                     200  0.3249067      2 0.4820036</span>
<span class="co">#&gt; 3                  67                      67  0.4425462      1 0.4919781</span>
<span class="co">#&gt; 4                  67                      67  0.4425462      2 0.5080219</span>
<span class="co">#&gt; 5                  66                      66  0.4570745      1 0.5312395</span>
<span class="co">#&gt; 6                  66                      66  0.4570745      2 0.4687605</span>
<span class="co">#&gt; 7                  67                      67  0.3942638      1 0.5309012</span>
<span class="co">#&gt; 8                  67                      67  0.3942638      2 0.4690988</span>
<span class="co">#&gt;   haplofreq.lower haplofreq.upper reference   RR.est.  RR.lower RR.upper</span>
<span class="co">#&gt; 1       0.4683542       0.5664903       ref 1.0000000 1.0000000 1.000000</span>
<span class="co">#&gt; 2       0.4335097       0.5316458        -  0.8788893 0.5972586 1.300834</span>
<span class="co">#&gt; 3       0.4080047       0.5767633       ref 1.0000000 1.0000000 1.000000</span>
<span class="co">#&gt; 4       0.4232367       0.5919953        -  1.4932158 0.6929969 3.201399</span>
<span class="co">#&gt; 5       0.4447807       0.6145346       ref 1.0000000 1.0000000 1.000000</span>
<span class="co">#&gt; 6       0.3854654       0.5552193        -  0.7923367 0.4070791 1.554546</span>
<span class="co">#&gt; 7       0.4449059       0.6127846       ref 1.0000000 1.0000000 1.000000</span>
<span class="co">#&gt; 8       0.3872154       0.5550941        -  0.6588865 0.3463119 1.259176</span>
<span class="co">#&gt;   RR.p.value RRdd.est. RRdd.lower RRdd.upper RRdd.p.value</span>
<span class="co">#&gt; 1         NA 1.0000000  1.0000000   1.000000           NA</span>
<span class="co">#&gt; 2  0.5169169 1.1562777  0.6731963   1.979085    0.6052218</span>
<span class="co">#&gt; 3         NA 1.0000000  1.0000000   1.000000           NA</span>
<span class="co">#&gt; 4  0.2970941 1.8696232  0.6871265   5.107933    0.2111747</span>
<span class="co">#&gt; 5         NA 1.0000000  1.0000000   1.000000           NA</span>
<span class="co">#&gt; 6  0.4973229 1.1586112  0.4549026   2.909822    0.7646181</span>
<span class="co">#&gt; 7         NA 1.0000000  1.0000000   1.000000           NA</span>
<span class="co">#&gt; 8  0.2030811 0.8024719  0.3125718   2.007631    0.6365470</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">g_x_me_results</span><span class="op">)</span></code></pre></div>
<p><img src="Run_analysis_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
<p>To check the significance of the interaction, we need to use <code>gxe</code> function</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gxe</span><span class="op">(</span><span class="va">g_x_me_results</span><span class="op">)</span>
<span class="co">#&gt;           gxe.test     chisq df      pval</span>
<span class="co">#&gt; 1       haplo.freq 0.5026134  2 0.7777838</span>
<span class="co">#&gt; 2            child 2.9488102  4 0.5664281</span>
<span class="co">#&gt; 3 haplo.freq.trend 0.3731476  1 0.5412935</span>
<span class="co">#&gt; 4      child.trend 2.6559459  2 0.2650139</span></code></pre></div>
<p>The p-values here show that the interaction was not significant.</p>
</div>
<div id="parent-of-origin-methylation-interactions" class="section level2">
<h2 class="hasAnchor">
<a href="#parent-of-origin-methylation-interactions" class="anchor"></a>Parent-of-origin-methylation interactions</h2>
<p>Similarly, we can also check for interaction between the methylation level and parent-of-origin effect of the genetic marker.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">poo_x_me_results</span> <span class="op">&lt;-</span> <span class="fu">haplinStrat</span><span class="op">(</span>
  data <span class="op">=</span> <span class="va">gen_ex_strat_prep</span>,
  strata <span class="op">=</span> <span class="fl">2</span>,
  poo <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span>
<span class="co">#&gt; Warning: Can only (for the time being) use reference = "ref.cat" or "population"</span>
<span class="co">#&gt; when poo == TRUE. Has been changed to "ref.cat".</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ## Running haplinStrat ##</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Selected stratification variable: dnam_c.c</span>
<span class="co">#&gt; Frequency distribution of stratification variable:</span>
<span class="co">#&gt;  1  2  3 </span>
<span class="co">#&gt; 67 66 67 </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Running Haplin on full data file...Done</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Running Haplin on stratum "1"...Done</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Running Haplin on stratum "2"...Done</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Running Haplin on stratum "3"...Done</span>
<span class="fu">haptable</span><span class="op">(</span><span class="va">poo_x_me_results</span><span class="op">)</span>
<span class="co">#&gt;   stratum row.str marker alleles  counts    HWE.pv Original After.rem.NA</span>
<span class="co">#&gt; 1     all       1     m1     1/2 607/593 0.3708224      200          200</span>
<span class="co">#&gt; 2     all       2   &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;        NA      200          200</span>
<span class="co">#&gt; 3       1       1     m1     1/2 178/224 0.4912064       67           67</span>
<span class="co">#&gt; 4       1       2   &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;        NA       67           67</span>
<span class="co">#&gt; 5       2       1     m1     1/2 206/190 0.6861466       66           66</span>
<span class="co">#&gt; 6       2       2   &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;        NA       66           66</span>
<span class="co">#&gt; 7       3       1     m1     1/2 223/179 0.1415450       67           67</span>
<span class="co">#&gt; 8       3       2   &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;        NA       67           67</span>
<span class="co">#&gt;   After.rem.Mend.inc. After.rem.unused.haplos pv.overall haplos haplofreq</span>
<span class="co">#&gt; 1                 200                     200  0.4215675      1 0.5176676</span>
<span class="co">#&gt; 2                 200                     200  0.4215675      2 0.4823324</span>
<span class="co">#&gt; 3                  67                      67  0.6427490      1 0.4930795</span>
<span class="co">#&gt; 4                  67                      67  0.6427490      2 0.5069205</span>
<span class="co">#&gt; 5                  66                      66  0.6552084      1 0.5308702</span>
<span class="co">#&gt; 6                  66                      66  0.6552084      2 0.4691298</span>
<span class="co">#&gt; 7                  67                      67  0.4587525      1 0.5302072</span>
<span class="co">#&gt; 8                  67                      67  0.4587525      2 0.4697928</span>
<span class="co">#&gt;   haplofreq.lower haplofreq.upper reference RRcm.est. RRcm.lower RRcm.upper</span>
<span class="co">#&gt; 1       0.4700792       0.5670512       ref 1.0000000  1.0000000   1.000000</span>
<span class="co">#&gt; 2       0.4329488       0.5299208        -  0.9544270  0.6137472   1.511417</span>
<span class="co">#&gt; 3       0.4078670       0.5757597       ref 1.0000000  1.0000000   1.000000</span>
<span class="co">#&gt; 4       0.4242403       0.5921330        -  1.5606351  0.6593474   3.661163</span>
<span class="co">#&gt; 5       0.4479463       0.6150836       ref 1.0000000  1.0000000   1.000000</span>
<span class="co">#&gt; 6       0.3849164       0.5520537        -  0.8267190  0.3738303   1.870074</span>
<span class="co">#&gt; 7       0.4480859       0.6138420       ref 1.0000000  1.0000000   1.000000</span>
<span class="co">#&gt; 8       0.3861580       0.5519141        -  0.7716568  0.3761164   1.625104</span>
<span class="co">#&gt;   RRcm.p.value RRcf.est. RRcf.lower RRcf.upper RRcf.p.value RRcm_RRcf.est.</span>
<span class="co">#&gt; 1           NA 1.0000000  1.0000000   1.000000           NA       1.000000</span>
<span class="co">#&gt; 2    0.8579539 0.7953424  0.4995210   1.289259    0.3484999       1.202425</span>
<span class="co">#&gt; 3           NA 1.0000000  1.0000000   1.000000           NA       1.000000</span>
<span class="co">#&gt; 4    0.3100549 1.4407954  0.6019219   3.440700    0.4240687       1.080438</span>
<span class="co">#&gt; 5           NA 1.0000000  1.0000000   1.000000           NA       1.000000</span>
<span class="co">#&gt; 6    0.6510574 0.7508834  0.3337436   1.712289    0.5004946       1.100715</span>
<span class="co">#&gt; 7           NA 1.0000000  1.0000000   1.000000           NA       1.000000</span>
<span class="co">#&gt; 8    0.4954824 0.5381458  0.2415788   1.232018    0.1409641       1.432857</span>
<span class="co">#&gt;   RRcm_RRcf.lower RRcm_RRcf.upper RRcm_RRcf.p.value RRdd.est. RRdd.lower</span>
<span class="co">#&gt; 1       1.0000000        1.000000                NA 1.0000000  1.0000000</span>
<span class="co">#&gt; 2       0.7361186        1.981824         0.4619789 1.1487113  0.6781981</span>
<span class="co">#&gt; 3       1.0000000        1.000000                NA 1.0000000  1.0000000</span>
<span class="co">#&gt; 4       0.4823310        2.468996         0.8399792 1.8820947  0.6829190</span>
<span class="co">#&gt; 5       1.0000000        1.000000                NA 1.0000000  1.0000000</span>
<span class="co">#&gt; 6       0.4520536        2.759066         0.8271603 1.1445511  0.4635546</span>
<span class="co">#&gt; 7       1.0000000        1.000000                NA 1.0000000  1.0000000</span>
<span class="co">#&gt; 8       0.6120656        3.383351         0.4057060 0.7890665  0.3166753</span>
<span class="co">#&gt;   RRdd.upper RRdd.p.value</span>
<span class="co">#&gt; 1    1.00000           NA</span>
<span class="co">#&gt; 2    2.00181    0.5993433</span>
<span class="co">#&gt; 3    1.00000           NA</span>
<span class="co">#&gt; 4    4.94472    0.2225700</span>
<span class="co">#&gt; 5    1.00000           NA</span>
<span class="co">#&gt; 6    2.99761    0.7579938</span>
<span class="co">#&gt; 7    1.00000           NA</span>
<span class="co">#&gt; 8    2.05191    0.6357477</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">poo_x_me_results</span><span class="op">)</span></code></pre></div>
<p><img src="Run_analysis_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="fu">gxe</span><span class="op">(</span><span class="va">poo_x_me_results</span><span class="op">)</span>
<span class="co">#&gt;           gxe.test     chisq df      pval</span>
<span class="co">#&gt; 1       haplo.freq 0.5026134  2 0.7777838</span>
<span class="co">#&gt; 2              poo 0.2626489  2 0.8769332</span>
<span class="co">#&gt; 3 haplo.freq.trend 0.3731476  1 0.5412935</span>
<span class="co">#&gt; 4        poo.trend 0.2175134  1 0.6409412</span></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Julia Romanowska, Haakon K. Gjessing.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
